godtools/App/Features/Lessons/Presentation/Lessons/LessonsViewModel.swift

```diff
@@ -42,18 +42,21 @@ class LessonsViewModel: ObservableObject {
         
         getCurrentAppLanguageUseCase
             .getLanguagePublisher()
+            .receive(on: DispatchQueue.main)
             .assign(to: &$appLanguage)
         
         $appLanguage.eraseToAnyPublisher()
-            .flatMap({ (appLanguage: AppLanguageDomainModel) -> AnyPublisher<ViewLessonsDomainModel, Never> in
+            .dropFirst()
+            .map { (appLanguage: AppLanguageDomainModel) in
             
                 return viewLessonsUseCase
                     .viewPublisher(appLanguage: appLanguage)
                     .eraseToAnyPublisher()
-            })
+            }
+            .switchToLatest()
```

Comment:
The main reasoning behind switchToLatest is from what I read in their discussion here (https://developer.apple.com/documentation/combine/publisher/switchtolatest()-453ht#discussion)

"This operator works with an upstream publisher of publishers, flattening the stream of elements to appear as if they were coming from a single stream of elements. It switches the inner publisher as new ones arrive but keeps the outer publisher constant for downstream subscribers."

"When this operator receives a new publisher from the upstream publisher, it cancels its previous subscription. Use this feature to prevent earlier publishers from performing unnecessary work, such as creating network request publishers from frequently updating user interface publishers."

With that, I really think we would benefit using switchToLatest in our ViewModels in situations where we are observing changes on a Published property that is then operating on another Publisher (UseCase).  This is because when the Published property value changes, it would then cancel any work happening in the UseCase and start new work on the UseCase.

It also seems to ignore all of the inner Realm objectWillChange notifications ("flattening the stream") and only takes the latest notification produced.

So I think a two-fold benefit of using switchToLatest.  It should really help in reducing a ton of overhead too.